tools/schemas/schema.ts

```diff
@@ -11,4 +11,46 @@ const MonorepoSchema = z.object({
   patternGroups: UrlSchema,
 });
 
-export { MonorepoSchema };
+const PackageRuleSchema = z.object({
+  matchCurrentVersion: z.string().optional(),
+  matchDatasources: z.array(z.string()),
+  matchPackageNames: z.array(z.string()),
+  replacementName: z.string().optional(),
+  replacementVersion: z.string().optional(),
+  description: z.string().optional(),
+  replacementNameTemplate: z.string().optional(),
+});
+
+const RuleSetSchema = z.object({
+  description: z.string(),
+  packageRules: z
+    .array(PackageRuleSchema)
+    .min(1)
+    .refine(
+      (rules) =>
+        rules.some(
+          (rule) =>
+            rule.replacementName !== undefined ||
+            rule.replacementNameTemplate !== undefined,
+        ),
+      {
+        message:
+          'At least one package rule must have either replacementName or replacementNameTemplate',
```

Comment:
```suggestion
          'At least one package rule must use either the replacementName config option, or the replacementNameTemplate config option',
```

How about something like this?